### ISSUE TITLE
error subsetting a section by pressure

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1238]
```R
library(oce)
data(section)
s <- subset(section, pressure > 100)
```
yields
```
Error in x@data$station[[i]]@data[r, ] : incorrect number of dimensions
```

**Comment by dankelley** [comment]
Fixed in `develop` branch, commit 8961f2b0793ea4f60733fa84dd717eae02f3b746

**Comment by richardsc** [comment]
Because the issue title is general enough, I'm going to reopen this for a similar-looking case I just discovered while testing a bunch of your recent changes.

I tried to subset for a *maximum* pressure, e.g. limit the `section` object to 2000 dbar:
```r
library(oce)
data(section)
plot(subset(section, pressure < 2000), which='temperature')
```
But it seems to do nothing ...
![image](https://cloud.githubusercontent.com/assets/233584/26039553/76124eec-38f2-11e7-91bb-35e88ee28c6f.png)

**Comment by dankelley** [comment]
Thanks -- I must have broken that when I permitted the `indices` argument to be a logical value.  (I fiddled with some code then.)  I'll fix it this afternoon, and add tests to the test suite, so this bug won't be able to return.  (I will report the file and line number in the test suite, when I write to say this is fixed.)

**Comment by dankelley** [comment]
Oh, I see the problem: R has a built-in dataset named `pressure`, and the code as updated on the weekend is interpreting the logical in the *outside* environment (this where-to-evaluate notion is difficult to explain, so I won't bother).

I've fixed the problem in `develop` commit f2e6b9aef6a6373226fe76cfde025319793bff75 and added a test in commit a862b117172a94a8eb3e5976d90e012eb2751fef

Below is how it works in now. Also, I added tests to `tests/testthat/test_section.R` starting in line 89, and these should ensure that the problem does not crop up again. One of the tests is "visual", meaning that it doesn't require much R skill to read and mimic. The other test is more subtle, including using `rownames` so that `testthat::expect_equal`` can be used to ensure identical data frames.
```R
> data(section)
> ss<-subset(section, pressure<2000)
> as.data.frame(section[["station",100]]@data)
   pressure temperature salinity salinityBottle oxygen silicate nitrite NO2.NO3 phosphate
1      11.4     23.5541  36.4766        36.4861  213.6     1.81    0.00    0.00      0.00
2      98.1     21.5704  36.6921        36.7017  204.9     1.71    0.08    0.45      0.01
3     192.9     18.8861  36.6001        36.6108  206.6     1.97    0.00    1.96      0.08
4     381.0     18.2375  36.5399        36.5433  199.2     2.49    0.02    3.88      0.16
5     598.8     16.3989  36.2388        36.2267  177.5     4.59    0.01    8.86      0.45
6     777.1     13.3255  35.7580        35.7346  168.4     8.51      NA   15.28      0.82
7     926.5      9.6363  35.2765        35.2333  144.9    16.78      NA   23.73      1.39
8    1189.5      5.8429  35.0700        35.0698  217.6    15.98      NA   20.55      1.23
9    1590.1      4.2909  34.9622        34.9797  264.6    14.04      NA   17.69      1.07
10   1699.8      4.0992  34.9497        34.9657  269.8    14.03      NA   17.36      1.17
11   1859.5      3.9220  34.9484        34.9543  272.4    14.28      NA   17.01      1.03
12   2081.2      3.8358  34.9639        34.9649  271.9    15.49      NA   17.33      1.05
13   2292.5      3.6493  34.9395        34.9549  275.4    16.00      NA   17.23      1.03
14   2482.8      3.5519  34.9645        34.9570  275.4    17.20      NA   17.05      1.05
15   2690.4      3.3836  34.9665        34.9545  272.3    18.41      NA   17.17      1.05
16   2895.6      3.1853  34.9516        34.9507  274.9    20.13      NA   17.14      1.06
17   3114.1      2.9787  34.9408        34.9405  274.8    21.50      NA   17.12      1.05
18   3316.3      2.7601  34.9307        34.9275  276.5    22.19      NA   16.71      1.05
19   3530.9      2.5878  34.9193        34.9168  278.7    23.55      NA   16.79      1.05
20   3746.1      2.4308  34.9040        34.9101  282.1    24.57      NA   16.84      1.07
21   3970.6      2.3322  34.8960        34.9007  278.8    28.25      NA   17.20      1.11
22   4189.5      2.2805  34.8838        34.8960  277.2    31.38      NA   17.74      1.13
23   4346.7      2.2514  34.8834        34.8920  275.0    33.08      NA   17.37      1.15
24   4398.3      2.2233  34.9039        34.8887  273.3    36.52      NA   18.40      1.16
> as.data.frame(ss[["station",100]]@data)
   pressure temperature salinity salinityBottle oxygen silicate nitrite NO2.NO3 phosphate
1      11.4     23.5541  36.4766        36.4861  213.6     1.81    0.00    0.00      0.00
2      98.1     21.5704  36.6921        36.7017  204.9     1.71    0.08    0.45      0.01
3     192.9     18.8861  36.6001        36.6108  206.6     1.97    0.00    1.96      0.08
4     381.0     18.2375  36.5399        36.5433  199.2     2.49    0.02    3.88      0.16
5     598.8     16.3989  36.2388        36.2267  177.5     4.59    0.01    8.86      0.45
6     777.1     13.3255  35.7580        35.7346  168.4     8.51      NA   15.28      0.82
7     926.5      9.6363  35.2765        35.2333  144.9    16.78      NA   23.73      1.39
8    1189.5      5.8429  35.0700        35.0698  217.6    15.98      NA   20.55      1.23
9    1590.1      4.2909  34.9622        34.9797  264.6    14.04      NA   17.69      1.07
10   1699.8      4.0992  34.9497        34.9657  269.8    14.03      NA   17.36      1.17
11   1859.5      3.9220  34.9484        34.9543  272.4    14.28      NA   17.01      1.03
```

**Comment by richardsc** [comment]
Ha, good thing I checked! Looks good now. Closing
